db/index.js

```diff
@@ -22,41 +26,47 @@ const useJson = (() => {
 var DB = (() => {
 
   const validateCode = async (code) => {
-    code = code.toUpperCase()
 
     if (useJson) {
-      return await jsonDB.find(user => user.code === code) || null
+      return await jsonDB.find(user => verifyHash(code.toUpperCase(), user.code, true)) || null
     }
 
+    code = hashString(code, true).catch(e => console.warn(e))
```

Comment:
What are the scenarios that could cause an error, just an empty string at this point?

Reply:
Yeah, basically. Which is not likely to happen, but just in case we've got there. We also reject the promise in hashSTring, and if we don't have a `catch` here then it's flagged as an unhandled promise rejection which is deprecated (and in the future will just exit our app).